lib/iris/tests/unit/experimental/ugrid/test_MeshCoord.py

```diff
@@ -0,0 +1,394 @@
+        mesh1 = _create_test_mesh()
+        mesh2 = _create_test_mesh()  # Presumably identical, but not the same
+        meshcoord1 = _create_test_meshcoord(mesh=mesh1)
+        meshcoord2 = _create_test_meshcoord(mesh=mesh2)
+        # These should NOT compare, because the Meshes are not identical : at
+        # present, Mesh equality is not implemented (i.e. limited to identity)
+        self.assertNotEqual(meshcoord2, meshcoord1)
+
+    def test_different_location(self):
+        meshcoord = self._create_common_mesh()
+        meshcoord2 = self._create_common_mesh(location="node")
+        self.assertNotEqual(meshcoord2, meshcoord)
+
+    def test_different_axis(self):
+        meshcoord = self._create_common_mesh()
+        meshcoord2 = self._create_common_mesh(axis="y")
+        self.assertNotEqual(meshcoord2, meshcoord)
+
+
+class Test__copy(tests.IrisTest):
+    def test_basic(self):
+        meshcoord = _create_test_meshcoord()
+        meshcoord2 = meshcoord.copy()
+        self.assertIsNot(meshcoord2, meshcoord)
+        self.assertEqual(meshcoord2, meshcoord)
+        # In this case, they should share *NOT* copy the Mesh object.
+        self.assertIs(meshcoord2.mesh, meshcoord.mesh)
+
+    def test_fail_copy_newpoints(self):
+        meshcoord = _create_test_meshcoord()
+        with self.assertRaisesRegex(ValueError, "Cannot change the content"):
+            meshcoord.copy(points=meshcoord.points)
+
+    def test_fail_copy_newbounds(self):
+        meshcoord = _create_test_meshcoord()
+        with self.assertRaisesRegex(ValueError, "Cannot change the content"):
+            meshcoord.copy(bounds=meshcoord.bounds)
+
+
+class Test__getitem__(tests.IrisTest):
+    def test_slice_wholeslice_1tuple(self):
+        # The only slicing case that we support, to enable cube slicing.
+        meshcoord = _create_test_meshcoord()
+        meshcoord2 = meshcoord[
+            :,
+        ]
+        self.assertIsNot(meshcoord2, meshcoord)
+        self.assertEqual(meshcoord2, meshcoord)
+        # In this case, we should *NOT* copy the linked Mesh object.
+        self.assertIs(meshcoord2.mesh, meshcoord.mesh)
+
+    def test_slice_whole_slice_singlekey(self):
+        # A slice(None) also fails, if not presented in a 1-tuple.
+        meshcoord = _create_test_meshcoord()
+        with self.assertRaisesRegex(ValueError, "Cannot index"):
+            meshcoord[:]
+
+    def test_fail_slice_part(self):
+        meshcoord = _create_test_meshcoord()
+        with self.assertRaisesRegex(ValueError, "Cannot index"):
+            meshcoord[:1]
+
+
+class Test_cube_containment(tests.IrisTest):
+    # Check that we can put a MeshCoord into a cube, and have it behave just
+    # like a regular AuxCoord.
+    def setUp(self):
+        meshcoord = _create_test_meshcoord()
+        data_shape = (2,) + _TEST_POINTS.shape
+        cube = Cube(np.zeros(data_shape))
+        cube.add_aux_coord(meshcoord, 1)
```

Comment:
I guess if this goes wrong, it goes wrong here in the `setup` :+1:

Reply:
Yes, that's just why `test_basic` appears to be a bit "thin" !